### ISSUE TITLE
[Feature Request] Delay between spinner start and finish

### ISSUE THREAD

**Issue by lonix1** [Napster2210/ngx-spinner#97]
Suppose we have this (simple example, in reality we would use promises or rxjs):

````
spinner.show();
doSomething();
spinner.show();
````

If `doSomething()` finishes quickly, then the spinner isn't seen at all.

How can we force the spinner to start, delay for 2 seconds, and then do everything else?

Maybe there is an option in this library for "delayed start" or something like that?

**Comment by Napster2210** [comment]
@Ionix1 You want to perform some action after spinner start but before end? Can you please give me more details? One more thing both the methods `show()` and `hide()` are already promises, if it works in your case.

**Comment by lonix1** [comment]
I do something like this:

````typescript
spinner.show()
  .then(x => new Promise(resolve => {
    setTimeout(() => resolve(x), 2000);
  }))
  .then(() => doSomething())
  .finally(() => spinner.hide());
````

This way, if `doSomething()` is very quick, the spinner will still be shown for 2 seconds. Sometimes that is important for UX, so the user knows that something important is happening.

But it would be very nice if it were available in the library itself, e.g. `spinner.showAndDelay(2000)` or something like that...

**Comment by Napster2210** [comment]
Okay got it, I'll try to add dynamic delay input in next release

**Comment by rspl-yuvraj** [comment]
@lonix1 This has been resolved and will be available in next version.

**Comment by Napster2210** [comment]
This is available in version `v8.1.0`

**Comment by lixaotec** [comment]
what is changed? where can I see this feature?
I wish to make ngx-spinner to only shows indeed after a second (x time) because sometimes, operations within 1 sec it starts and stops resulting in undesirebla effect of a spinner that happens too fast
